### PR TITLE
hopefully fixes unit test failures from things in the spatial grid not having their refs cleared from it

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
things being held inside the spatial grid while they should be deleted causes a rareish unit test failure for create_and_destroy. hopefully this fixes it by checking for if the entering movable is being qdeleted as well as early returning in enter_cell if the spatial grid isnt initialized. 

Im not sure what exactly causes the hanging refs, but i suspect its something to do with not properly catching qdeleting objects if they for whatever reason enter another part of the spatial grid instead of nullspacing

### CODE REVIEW

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 1]
```diff
@@ -349,7 +349,7 @@ SUBSYSTEM_DEF(spatial_grid)
 ///find the spatial map cell that target belongs to, then add target's important_recusive_contents to it.
 ///make sure to provide the turf new_target is "in"
 /datum/controller/subsystem/spatial_grid/proc/enter_cell(atom/movable/new_target, turf/target_turf)
-	if(!target_turf || !new_target?.important_recursive_contents)
+	if(!target_turf || QDELETING(new_target) || !new_target?.important_recursive_contents)
 		CRASH("/datum/controller/subsystem/spatial_grid/proc/enter_cell() was given null arguments or a new_target without important_recursive_contents!")
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/spatial_gridmap.dm:355]: Shouldn't this have its own CRASH?

Comment (Kylerace) [reply code/controllers/subsystem/spatial_gridmap.dm]: good idea

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 2]
```diff
@@ -349,7 +349,9 @@ SUBSYSTEM_DEF(spatial_grid)
 ///find the spatial map cell that target belongs to, then add target's important_recusive_contents to it.
 ///make sure to provide the turf new_target is "in"
 /datum/controller/subsystem/spatial_grid/proc/enter_cell(atom/movable/new_target, turf/target_turf)
-	if(!target_turf || !new_target?.important_recursive_contents)
+	if(!initialized)
+		return
+	if(!target_turf || QDELING(new_target) || !new_target?.important_recursive_contents)
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/spatial_gridmap.dm:354]: Don't we want QDELETED here?

Comment (Kylerace) [reply code/controllers/subsystem/spatial_gridmap.dm]: technically the null check is handled by !new_target?.important_recursive_contents but i made it QDELETED after i separated it into its own check

**Pull request merged**